src/mlpack/CMakeLists.txt

```
@@ -158,3 +159,80 @@ endif ()
 # If we are building Markdown documentation, we have to run some setup after we
 # recurse into methods/.  If not, this function is empty.
 post_markdown_setup()
+
+# Installing CMake Helper
+
+# Joins arguments and places the results in ${result_var}.
+function(join result_var)
+set(result )
+foreach (arg ${ARGN})
+  set(result "${result}${arg}")
+endforeach ()
+set(${result_var} "${result}" PARENT_SCOPE)
+endfunction()
+
+# Sets a cache variable with a docstring joined from multiple arguments:
+#   set(<variable> <value>... CACHE <type> <docstring>...)
+# This allows splitting a long docstring for readability.
+function(set_verbose)
+cmake_parse_arguments(SET_VERBOSE "" "" "CACHE" ${ARGN})
+list(GET SET_VERBOSE_CACHE 0 type)
+list(REMOVE_AT SET_VERBOSE_CACHE 0)
+join(doc ${SET_VERBOSE_CACHE})
+set(${SET_VERBOSE_UNPARSED_ARGUMENTS} CACHE ${type} ${doc})
+endfunction()
+
+include(CMakePackageConfigHelpers)
+
+set_verbose(MLPACK_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/mlpack CACHE STRING
+          "Installation directory for cmake files, relative to "
+          "${CMAKE_INSTALL_PREFIX}.")
+set(version_config ${PROJECT_BINARY_DIR}/mlpack-config-version.cmake)
+set(project_config ${PROJECT_BINARY_DIR}/mlpack-config.cmake)
+set(targets_export_name mlpack-targets)
+
+set(INSTALL_TARGETS mlpack)
+
+set_verbose(MLPACK_LIBS_DIR ${CMAKE_INSTALL_LIBDIR} CACHE STRING
+          "Installation directory for libraries, relative to "
+          "${CMAKE_INSTALL_PREFIX}.")
+
+set_verbose(MLPACK_INCS_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
+          "Installation directory for include files, relative to "
+          "${CMAKE_INSTALL_PREFIX}.")
+
+# Getting the absolute path, for some reason CMAKE gets confused, 
+# even with CMAKE_INSTALL_INCLUDEDIR
+if (UNIX)
+GNUInstallDirs_get_absolute_install_dir(MLPACK_LIBS_DIR CMAKE_INSTALL_LIBDIR)
+GNUInstallDirs_get_absolute_install_dir(MLPACK_INCS_DIR CMAKE_INSTALL_INCLUDEDIR)
+endif()
+
+# Generate the version, config and target files into the build directory.
+write_basic_package_version_file(
+${version_config}
+VERSION ${PACKAGE_VERSION}
+COMPATIBILITY SameMajorVersion)
```

```suggestion
write_basic_package_version_file(
    ${version_config}
    VERSION ${PACKAGE_VERSION}
    COMPATIBILITY SameMajorVersion)
```
Just some formatting. :)  We don't have the style check job for CMake but I can point it out manually, at risk of being pedantic.  Sorry about that. :)